src/libYARP_dev/src/devices/RGBDSensorClient/RGBDSensorClient.cpp

```diff
@@ -300,6 +300,9 @@ bool RGBDSensorClient::close()
     colorFrame_StreamingPort.close();
     depthFrame_StreamingPort.close();
     rpcPort.close();
+    delete RgbMsgSender;
+    delete DepthMsgSender;
+    delete streamingReader;
```

Comment:
Since these are initialized in the constructor and not in the `open()` method, I think they should be deleted in the destructor instead of in the `close()` method